### PR TITLE
Add support for zilliz

### PR DESCRIPTION
**Pull request by EchoAGI**
I've added support for the vector database Zilliz, which is the cloud version of Milvus. There are some differences in details between Zilliz and the open-source version of Milvus. To accommodate production environments, we have put explicit naming on the collection field instead of the random field naming used by Milvus.

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
we already have a notebook with zilliz support - https://python.langchain.com/en/latest/modules/indexes/vectorstores/examples/zilliz.html

is this needed?

**Comment by EchoAGI** [comment]
> we already have a notebook with zilliz support - https://python.langchain.com/en/latest/modules/indexes/vectorstores/examples/zilliz.html
> 
> is this needed?

the example of the above document doesn't work, because some details are different between milvus and zilliz, such as index type, zilliz only support "AUTOINDEX", but milvus support "HNWS".

File: langchain/vectorstores/zilliz.py [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+            self.fields.append(x.name)
+            if x.auto_id:
+                self.fields.remove(x.name)
+            if x.is_primary:
+                self.primary_field = x.name
+            if x.dtype == DataType.FLOAT_VECTOR or x.dtype == DataType.BINARY_VECTOR:
+                self.vector_field = x.name
+
+        # Default search params when one is not provided.
+        self.index_params = {
```
[the remark this answers is not in the document]
Comment (EchoAGI) [reply langchain/vectorstores/zilliz.py:81]: I'm not sure what you mean

[the remark this answers is not in the document]
Comment (vowelparrot) [reply langchain/vectorstores/zilliz.py:81]: I mean - it says they're defaults when one is not provided but there doesn't seem to be a way to provide index params. A reader of this code may not know why the default of 10 is chosen for everything, so any links to explain would be helpful

[the remark this answers is not in the document]
Comment (EchoAGI) [reply langchain/vectorstores/zilliz.py:81]: oh I got it

**Comment by hwchase17** [comment]
closing in favor of other notebook

**Pull request closed without merging**